addons/website/models/website.py

```diff
@@ -1751,7 +1750,13 @@ def _trigram_enumerate_words(self, search_details, search, limit):
             best_similarity = sql.SQL('GREATEST({similarities})').format(
                 similarities=sql.SQL(', ').join(similarities)
             )
-
+            # This method uses hardcoded sql query, it needs filter regarding performance.
+            # In this case for the is_published field as customer don't have access to
+            # unpublished objects.
+            if 'is_published' in model._fields and not self.env.user.has_group('base.group_user'):
+                where_clause = sql.SQL("WHERE is_published")
```

Comment:
This could crash, `model._fields` could return fields from another table, through the `inherits` mechanism.
For instance, if you do `'active' in self.env['website.page']._fields`, it will be true, but if you do `WHERE active` against the `website_page` table, it will crash as this field doesn't exists, it exists in the `ir_ui_view` table, since `website.page` inherit**s** from `ir.ui.view`.
```suggestion
            if 'is_published' in model._model_fields and not self.env.user.has_group('base.group_user'):
                where_clause = sql.SQL("WHERE is_published")
```
To test, I just wrote that on github

Reply:
If inherited field it is not stored by default 

https://github.com/odoo/odoo/blob/41c19242348b17b6558f23a5d338d8bd06fd9ca5/odoo/models.py#L3158